--- a/scripts/snmpsimd.py	2019-11-03 15:02:07.398220874 +0000
+++ b/scripts/snmpsimd.py	2019-11-03 15:02:46.768884664 +0000
@@ -1599,12 +1599,13 @@
 
                     dataIndexInstrumController = DataIndexInstrumController()
 
-                    configureManagedObjects(
-                        ctxDataDirs or dataDirs or confdir.data, 
-                        dataIndexInstrumController,
-                        snmpEngine,
-                        snmpContext
-                    )
+                    with daemon.PrivilegesOf(procUser, procGroup):
+                        configureManagedObjects(
+                            ctxDataDirs or dataDirs or confdir.data, 
+                            dataIndexInstrumController,
+                            snmpEngine,
+                            snmpContext
+                        )
 
                 # Configure access to data index
 
